src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/AzureSqlVMCmdletBase.cs

```diff
@@ -0,0 +1,99 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Model;
+using Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Adapter;
+using System.Collections.Generic;
+using Microsoft.Azure.Commands.SqlVirtualMachine.Common;
+using System.Management.Automation;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Cmdlet
+{
+    /// <summary>
+    /// Generic cmdlet for operations that run over a sql virtual machine. 
+    /// </summary>
+    public abstract class AzureSqlVMCmdletBase : AzureSqlVirtualMachineCmdletBase<IEnumerable<AzureSqlVMModel>, AzureSqlVMAdapter>
+    {
+        /// <summary>
+        /// Intializes the model adapter
+        /// </summary>
+        /// <returns>The Sql virtual machine adapter</returns>
+        protected override AzureSqlVMAdapter InitModelAdapter()
+        {
+            return new AzureSqlVMAdapter(DefaultContext);
+        }
+
+        /// <summary>
+        /// Validate the resource id of a sql virtual machine
+        /// </summary>
+        /// <param name="sqlVirtualMachineResourceId">Resource id to be validated</param>
+        /// <returns>True if the resource id is valid, false otherwise</returns>
+        public bool ValidateSqlVirtualMachineId(string sqlVirtualMachineResourceId)
+        {
+            var regex = new Regex(@"/subscriptions/([^/]+)/resourceGroups/([^/]+)/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines/([^/]+)");
```

Comment:
It should be SqlVirtualMachines instead of sqlVirtualMachines.
Should we consider doing an case insensitive match?